cola/models/stash.py

```diff
@@ -104,9 +107,36 @@ def __init__(self, context, stash_name, keep_index):
 
     def do(self):
         if self.keep_index:
-            args = ['save', '--keep-index', self.stash_name]
+            args = ['push', '--keep-index', '-m', self.stash_name]
         else:
-            args = ['save', self.stash_name]
+            args = ['push', '-m', self.stash_name]
+        status, out, err = self.git.stash(*args)
+        if status == 0:
+            Interaction.log_status(status, out, err)
+        else:
+            title = N_('Error')
+            cmdargs = core.list2cmdline(args)
+            Interaction.command_error(
+                title, 'git stash ' + cmdargs, status, out, err)
+
+        self.model.update_status()
+
+
+class RenameStash(cmds.ContextCommand):
+    """Rename the stash"""
+
+    def __init__(self, context, stash_index, stash_name):
+        super(RenameStash, self).__init__(context)
+        self.context = context
+        self.stash_index = stash_index
+        self.stash_name = stash_name
+
+    def do(self):
+        # Drop the stash first and get the returned ref
+        ref = DropStash(self.context, self.stash_index).do()
+
+        # Store the stash with a new name
+        args = ['store', '-m', self.stash_name, ref]
```

Comment:
Ditto, if we wanted to be maximally compatible we can add a similar `stash-store` check for `2.2.0`, but only if there's a way to do this with the old `git stash save` here too.  We try to be maximally compatible.

Reply:
From what I see, `git stash store -m <message> <commit>` is available since git 1.8.4. Unfortunately, I don't see any command to achieve the goal before this version… Should we check the version and gray out/hide the rename button for versions prior to 1.8.4?

Reply:
Hiding the rename button for older versions makes sense.  Thanks for considering compatibility.